pliers/extractors/base.py

```diff
@@ -132,98 +170,103 @@ def history(self):
+                  and error).
+                - True: When format='long', behaves like 'column'. When
+                  format='wide', behaves like 'prepend'.
+
+        aggfunc (str, Callable): If format='wide' and extractor_names='drop',
+            it's possible for name clashes between features to occur. In such
+            cases, the aggfunc argument is passed onto pandas' pivot_table
+            function, and specifies how to aggregate multiple values for the
+            same index. Can be a callable or any string value recognized by
+            pandas. By default (None), 'mean' will be used for numeric columns
+            and 'first' will be used for object/categorical columns.
+
+    Returns: a pandas DataFrame. For format details, see 'format' argument.
     '''
 
-    # Flatten list recursively
-    results = flatten(results)
-
-    stims = defaultdict(list)
-
-    for r in results:
-        stims[hash(r.stim)].append(r)
-
-    # First concatenate all features separately for each Stim
-    for k, v in stims.items():
-        stims[k] = ExtractorResult.merge_features(v, **merge_feature_args)
-
-    # Now concatenate all Stims
-    stims = list(stims.values())
-    return stims[0] if len(stims) == 1 else \
-        ExtractorResult.merge_stims(stims)
+    _timing = True if timing == 'auto' else timing
+
+    if extractor_names is True:
+        extractor_names = 'prepend' if format == 'wide' else 'column'
+
+    dfs = [r.to_df(timing=_timing, metadata=metadata, format='long',
+                   extractor_name=True, add_object_id=True)
+           for r in results]
+
+    data = pd.concat(dfs, axis=0).reset_index(drop=True)
+
+    if extractor_names in ['prepend', 'multi']:
+        data['feature'] = data['extractor'] + '#' + data['feature'].astype(str)
+
+    if extractor_names != 'column':
+        data = data.drop('extractor', axis=1)
+
+    if format == 'wide':
+        ind_cols = {'stim_name', 'onset', 'duration', 'object_id', 'class',
+                    'filename', 'history', 'source_file'}
+        ind_cols = list(ind_cols & set(data.columns))
+        # pandas groupby/index operations can't handle NaNs in index, (see
+        # issue at https://github.com/pandas-dev/pandas/issues/3729), so we
+        # replace NaNs with a placeholder and then re-substitute after
+        # pivoting.
+        dtypes = data[ind_cols].dtypes
+        data[ind_cols] = data[ind_cols].fillna('PlAcEholdER')
+
+        # Set default aggfunc based on column type, otherwise bad things happen
+        if aggfunc is None:
+            aggfunc = 'mean' if is_numeric_dtype(data['value']) else 'first'
+
+        data = data.pivot_table(index=ind_cols, columns='feature',
+                                values='value', aggfunc=aggfunc).reset_index()
+        data[ind_cols] = data[ind_cols].replace('PlAcEholdER', np.nan)
+        data[ind_cols] = data[ind_cols].astype(dict(zip(ind_cols, dtypes)))
+
+    if timing == 'auto' and 'onset' in data.columns:
```

Comment:
Would be good to note what this is doing in the argument doc

Reply:
Also not 100% sold on this but at least its configurable I suppose

Reply:
Would you prefer to always leave those columns in? My concern is that in a lot of cases, there's no timing information, and then the user always gets back `onset` and `duration` columns that are all `NaN`s, which is both unhelpful and confusing. If we make the default `False`, then we risk users not realizing they can get back timing information--which can be critical. I don't really like the `'auto'` business either, but I'm not sure I see a better alternative...

Reply:
Fair enough, we can see how it goes this way.

Reply:
Shouldn't we also include the `auto` functionality in the ExtractorResult `to_df` method? Between this and the `object_id` comment below, its kind of tedious to write units tests since its difficult to remember which columns are guaranteed and which aren't between the two methods.

Reply:
@tyarkoni thoughts on moving the `auto` functionality to the ExtractorResult `to_df` method?

Reply:
Oh, sorry. Didn't see this till just now. Yeah, probably. Will update.

Reply:
Oh, actually, I just realized that we can't pass the `add_object_id` arg from `merge_results` to `to_df`, because then we risk getting back DFs in `merge_results` (i.e., from different `ExtractorResult` objects) that have different columns (i.e., some with `object_id` and some without). We could of course handle this with another check for the column, but that seems inelegant. So probably I'll just keep the former approach of always requesting the `object_id` column when called from `merge_results`. I'll still add the `auto` option to `to_df` though.